plasmapy/formulary/densities.py

```diff
@@ -0,0 +1,56 @@
+"""Functions to calculate plasma density parameters."""
+__all__ = [
+    "plasma_critical_density",
+]
+
+import astropy.units as u
+
+from astropy.constants.si import e, eps0, m_e
+
+from plasmapy.utils.decorators import validate_quantities
+
+
+@validate_quantities(
+    omega_p={"can_be_negative": False},
+    validations_on_return={
+        "units": [u.m**-3],
+        "equivalencies": [
+            (u.F * u.kg * u.rad**2 / (u.C**2 * u.m * u.s**2), u.m**-3)
+        ],
+    },
+)
+def plasma_critical_density(omega_p: u.rad / u.s) -> u.m**-3:
+    r"""Calculate the plasma critical density.
+
+    Parameters
+    ----------
+    omega_p : `~astropy.units.Quantity`
+        The plasma frequency in units of angular frequency.
```

Comment:
Actually, this should be the radiation frequency. They are equal at the critical density, but conceptually usually you know the radiation frequency and then what to find out what density it will be absorbed at. I would go through and change this and the variable name throughout. Just 'omega' is fine.